### PR TITLE
[.NET][WPF][HTML] Update BackgroundImage implementation for spec changes 

### PR DESCRIPTION
**Pull request by almedina-ms**
* Updates **stretch** enum value for **cover** (BackgroundImageMode)
* Updates the enum **BackgroundImageMode** to **ImageFillMode**
* Renames the background image property **Mode** to **FillMode**
* Adds support for **verticalAlignment** and **horizontalAlignment** in the _cover_ scenario

As detailed here https://github.com/Microsoft/AdaptiveCards/issues/2794

Fixes this: https://github.com/Microsoft/AdaptiveCards/issues/2797

### Pre-check

As it updates enum and property names, the first check was for the project to compile (it does).
Also the tests were run and worked as expected though a couple had to be updated

### VerticalAlignment and HorizontalAlignment

The card Tests/BackgroundImage.VerticalAndHorizontalAlignment.json was added to check the correct alignment of the image in the containers and columns

WPF

This one's a little weird, according to the documentation, auto should use as much space as needed, so if the columnset width overflows it's "expected"? I need more info on the person who made the first column implementation
image

I removed the word background here so the horizontal alignment effect could be viewed
image

HTML
image

### All other cards

HTML
image

WPF
image

HTML
image

WPF
image

HTML
image

WPF
image

HTML
image

WPF
image

HTML
image

WPF
image

HTML
image

WPF
image

HTML
image

WPF
image

HTML
image

WPF
image

### CODE REVIEW

**Comment by andrewleader** [comment]
Epic work Alberto!!

@matthidinger for columns, the `auto` width shouldn't overflow past the width of the card, correct?

As seen below, the column contents on the right are too large, so right now in WPF the column all exceeds

img

I believe the correct "auto" column width implementation is... the first "auto" column gets to consume as much width as it wants (but can't consume more than the remaining width), then the next "auto" width gets to consume as much remaining width available, etc...

So in that case, the right-most column's width should be set to the remaining width, causing the text to wrap

**Comment by paulcam206** [comment]
@almedina-ms -- has this been updated for @andrewleader's comments above?

**Comment by almedina-ms** [comment]
I'm not sure if I should address this comment here as this PR is only for the background image spec update while the column issue has been there for some time. Created an issue to fix later [https://github.com/microsoft/AdaptiveCards/issues/2862](https://github.com/microsoft/AdaptiveCards/issues/2862)

---
In reply to: [490187118](https://github.com/microsoft/AdaptiveCards/pull/2806#issuecomment-490187118) [](ancestors = 490187118)

File: samples/Tests/BackgroundImage.VerticalAndHorizontalAlignment.json [hunk 1]
```diff
@@ -0,0 +1,148 @@
+{
```
Comment (paulcam206) [inline samples/Tests/BackgroundImage.VerticalAndHorizontalAlignment.json]: Need to run the UWPTestApp and add reference files to this PR :)

**Pull request merged**